appmap/test/test_detect_enabled.py

```diff
@@ -0,0 +1,194 @@
+            assert DetectEnabled.should_enable(recording_method) == True
+
+    recording_methods_as_false = {
+        "_".join(["APPMAP", "RECORD", recording_method.upper()]): "false"
+        for recording_method in RECORDING_METHODS
+    }
+
+    @patch.dict(os.environ, recording_methods_as_false)
+    def test_some__recording_method_disabled(self):
+        for recording_method in RECORDING_METHODS:
+            DetectEnabled.initialize()
+            assert DetectEnabled.should_enable(recording_method) == False
+
+
+class TestDetectEnabledFlask:
+    @staticmethod
+    def setup_method():
+        DetectEnabled.clear_cache()
+
+    def test__none(self):
+        DetectEnabled.initialize()
+        assert DetectEnabled.should_enable(None) == False
+
+    @patch.dict(os.environ, {"FLASK_APP": "app.py"})
+    def test__flask_app(self):
+        DetectEnabled.initialize()
+        assert DetectEnabled.should_enable(None) == False
+
+    @patch.dict(os.environ, {"FLASK_DEBUG": "0"})
+    def test__flask_debug_0(self):
+        DetectEnabled.initialize()
+        assert DetectEnabled.should_enable("requests") == False
+
+    @patch.dict(os.environ, {"FLASK_DEBUG": "1"})
+    def test__flask_debug_1(self):
+        DetectEnabled.initialize()
+        assert DetectEnabled.should_enable("requests") == True
+
+    @patch.dict(os.environ, {"FLASK_ENV": "production"})
+    def test__flask_env_production(self):
+        DetectEnabled.initialize()
+        assert DetectEnabled.should_enable("requests") == False
+
+    @patch.dict(os.environ, {"FLASK_ENV": "development"})
+    def test__flask_env_development(self):
+        DetectEnabled.initialize()
+        assert DetectEnabled.should_enable("requests") == True
+
+
+class TestDetectEnabledDjango:
+    @staticmethod
+    def setup_method():
+        DetectEnabled.clear_cache()
+
+    def test__none(self):
+        DetectEnabled.initialize()
+        assert DetectEnabled.should_enable(None) == False
+
+    def driver(
+        self,
+        data_dir,
+        monkeypatch,
+        django_settings_module,
+        basename_settings,
+        extra_settings_content,
+        expected,
+    ):
+        settings_content = (
+            """
+# If the SECRET_KEY isn't defined we get the misleading error message
+# CommandError: You must set settings.ALLOWED_HOSTS if DEBUG is False.
+SECRET_KEY = "3*+d^_kjnr2gz)4q2m(&&^%$p4fj5dk3%lz4pl3g4m-%6!gf&)"
+
+# Must set ROOT_URLCONF else we get
+# AttributeError: 'Settings' object has no attribute 'ROOT_URLCONF'
+ROOT_URLCONF = "app.urls"
+
+MIDDLEWARE = ["appmap.django.Middleware"]
+
+"""
+            + extra_settings_content
+            + """
+"""
+        )
+        monkeypatch.syspath_prepend(data_dir / "django")
+        monkeypatch.setenv("DJANGO_SETTINGS_MODULE", django_settings_module)
+        filename = data_dir / "django" / "app" / basename_settings
+        try:
+            file_write(str(filename), settings_content)
```

Comment:
You should use a `with` statement for this, and get rid of `file_write`:
https://github.com/getappmap/appmap-python/blob/5858681742e47e6841ecd32379c297fded9df4ef/appmap/_implementation/configuration.py#L234

Reply:
Done.  I would have kept `file_write`.  A function that writes in a file seems like a useful abstraction.  I'm surprised there's still no such function in Python while there is one in Ruby.